### ISSUE TITLE
allow-in-func option missed for no-did-update-set-state rule

### ISSUE THREAD

**Issue by playpauseandstop** [yannickcr/eslint-plugin-react#300]
https://github.com/yannickcr/eslint-plugin-react/commit/4c167ebc16c1593489b61886716c48cfd2044f2d introduced `allow-in-func` option only to `no-did-mount-set-state` rule, but not for `no-did-update-set-state`. 

So if `this.setState` called in arrow function inside of `componentDidUpdate` - ESLint failed on `no-did-update-set-state` rule. E.g,

```js
  componentDidUpdate() {
    fetch("/api/").then(toJson).then(json => {
      this.setState({data: json});
    });
  }
```

Is it possible to fix this and introduce `allow-in-func` to `no-did-update-set-state` rule, or there were some non obvious reasons, why `allow-in-func` option missed in the rule?

Related to #56

**Comment by yannickcr** [comment]
No particular reason for not having it for `componentDidUpdate`, it was just more common to need it for `componentDidMount`. We can totally add the same option for `no-did-update-set-state`.

**Comment by remitbri** [comment]
Documentation should be updated on the [rule page](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-did-update-set-state.md).
